fairscale/optim/utils.py

```diff
@@ -52,11 +53,8 @@ def broadcast_object(
 
     if dist.get_rank() == src_rank:
         # Emit data
-        buffer = io.BytesIO()
-        torch.save(obj, buffer)
-        data = bytearray(buffer.getbuffer())
-        length_tensor = torch.LongTensor([len(data)]).to(dist_device)
-        data_send_tensor = torch.ByteTensor(data).to(dist_device)
+        data_send_tensor = pyobject_to_tensor(obj).to(dist_device)
```

Comment:
ah interesting, I didn't know of this way ! Could you elaborate on the benefits vs. BytesIO ? cc @mannatsingh, because this code comes from Classy initially. Else if that works better in any way that's fine by me, the code is a little cleaner I think

Reply:
Ah yeah this looks like a function within fairsclae which does something similar to what we do in classy, but through pickle and numpy (https://github.com/facebookresearch/fairscale/blob/63f7796a72f9afbcba64f9bf0df11753ecc4558d/fairscale/nn/pipe/pipeline.py#L108). Although I don't think we're supposed to pickle tensors. So if obj contains tensors this might become problematic?

Reply:
Ok, I'm going to revert my changes to fairscale.optim

Reply:
Just mentioned this to @blefaudeux over chat - I think the only issue with pickling tensors is the device handling bit. 
So I guess my statement isn't completely true - "I don't think we're supposed to pickle tensors". If the code works with both GPUs and CPUs, there's no need to get rid of that. Just something to be mindful of :)